src/qibo/hamiltonians/hamiltonians.py

```diff
@@ -16,15 +14,42 @@ class Hamiltonian(hamiltonians.MatrixHamiltonian):
             or on ``tf.sparse`` for the tensorflow backend are also
             supported.
     """
+    def __init__(self, nqubits, matrix=None, backend=None):
+        if backend is None:
+            from qibo.backends import GlobalBackend
+            self.backend = GlobalBackend()
 
-    def __init__(self, nqubits, matrix):
-        if not (isinstance(matrix, K.tensor_types) or K.issparse(matrix)):
+        if not (isinstance(matrix, self.backend.tensor_types) or self.backend.issparse(matrix)):
```

Comment:
Regarding this, I have defined the attributes `tensor_types` and `numeric_types` in the numpy backend, because there were a few checks especially in the `Hamiltonian` and `SymbolicHamiltonian` classes. For both terms and symbols I have replaced those attributes using `np.ndarray` as you suggested below in order to avoid invoking the backend. Let me know if you think that we should keep those attributes.

Reply:
Yes, I think we can keep them. It is fine as it is now.